### PR TITLE
Feature/respawn

### PR DESCRIPTION
**Pull request by ZacharyTaylor**
Changed it so watchdog only operates if in base_station_mode. Do you have any major concerns with this style of detecting and resetting?

### CODE REVIEW

File: piksi_multi_rtk_gps/startup/rc.local [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Make sure that the script will "exit 0" on success or any other
+# value on error.
+#
+# In order to enable or disable this script just change the execution
+# bits.
+#
+# By default this script does nothing.
+
+#run rtk on startup
+su - tradr -c 'source ~/catkin_ws/devel/setup.bash; roslaunch ~/catkin_ws/src/mav_rtk_gps/piksi_multi_rtk_gps/startup/startup.launch &'
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_gps/startup/rc.local]: What about placing this in the `mav_eurathlon` folder? It has some explicit dependency on the setup for the challenge, like the username and path of the catkin workspace.

File: piksi_multi_rtk_gps/startup/startup.launch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<launch>
+  <node name="master_discovery" pkg="master_discovery_fkie" type="master_discovery" >
+  </node>
+
+  <node name="master_sync" pkg="master_sync_fkie" type="master_sync" output="screen">
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_gps/startup/startup.launch]: Same as above. `master_sync_fkie` is not a dependency of this package and I'd rather not include something that is not strictly needed by the driver.

**Comment by marco-tranzatto** [comment]
@ZacharyTaylor that's the "real" error sometimes appears in the driver: [piksi disconnected error](https://github.com/swift-nav/libsbp/blob/786b41304b485b6cb8fdd2ba8509e3135b66ce14/python/sbp/client/drivers/pyserial_driver.py#L81)

I am not sure in which part of the driver we should catch this exception ...

**Comment by ZacharyTaylor** [comment]
The disconnect error is a little frustrating. They correctly throw a IOError exception if it happens, however the serial driver is running in its own thread and nothing catches the error. This means the thread crashes, and as they never check on the thread no other part of the program is aware of the crash.

I saw three solutions
1) Fork the driver repo and have the thread let others know before crashing
2) Redirect pythons print to a variable and periodically go through it looking for this error.
3) Add a watchdog timer on our side.

I went with 3) as it was quick and easy, but to do it properly we probably should do 1).

**Pull request merged**